File: JavaScript/Level.2/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let queue = progresses.map((progress) => 100 - progress);
    let index = 0;
    while (index < queue.length) {
        let finishedWork = 0;
        let day = Math.ceil(queue[index] / speeds[index]);
        while (
            index < queue.length &&
            Math.ceil(queue[index] / speeds[index]) <= day
        ) {
            ++finishedWork;
            ++index;
        }
        answer.push(finishedWork);
    }
    return answer;
}

File: C++/Level.2/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> TIMES;
    int n = progresses.size();
    
    for (int i = 0; i < n; ++i) {
        int remain = 100 - progresses[i];
        int day = remain / speeds[i] + (remain % speeds[i] != 0);
        TIMES.push_back(day);
    }
    
    for (int i = 0; i < n;) {
		int workDay = TIMES[i];
        int cnt = 0;
        while (i < n && TIMES[i] <= workDay) {
        	++cnt;
            ++i;
        }
        answer.push_back(cnt);
    }
    
    return answer;
}